src/components/TodoForm.js

```diff
@@ -7,39 +7,39 @@ import { createTheme, ThemeProvider } from "@mui/material/styles";
 import { useState, useEffect } from "react";
 import { FormControl } from "@mui/material";
 import CheckboxList from "./CheckboxList";
+import useSWR from "swr";
+import LoadingSpinner from "./spinner";
 
 const url = "https://jsonplaceholder.typicode.com/todos";
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
 
 function TodoForm() {
+  const { data, isError, isLoading } = useSWR(url, fetcher);
   const [text, setText] = useState("");
   const [tasks, setTasks] = useState([]);
-  const [isLoading, setIsLoading] = useState(true);
-  const [isError, setIsError] = useState(false);
-  const [todos, setTodos] = useState("default todo");
-
-  const getTodos = async () => {
-    const response = await fetch(url);
-    const todos = await response.json();
-    setTodos(todos);
-    let temp = [];
-    todos.map((todo) => {
-      if (todo.id <= 10) {
-        temp.push(todo);
-      }
-    });
-    setTasks(temp);
-    setIsLoading(false);
-  };
 
   useEffect(() => {
-    setIsLoading(true);
-    getTodos();
-  }, []);
+    setTasks(data ? data.slice(0, 10) : []);
+  }, [data]);
+
+  if (isError) {
+    <h1>errooooor - not found</h1>;
+  }
+  if (isLoading) {
+    return (
+      <div>
+        <LoadingSpinner />
+      </div>
+    );
+  }
 
   const Tasks = (text) => {
```

Comment:
Another recommendation about naming things: When naming a function try to choose a name that closely reflect what the function does.

For this particular case, `Task` doesn't say too much so other devs will be forced to read the implementation details. Something like `addNewTask`, `addTask` are probably better options. The same apply for the arguments of the function.

Here is an example of a better signature for this function:
`const addTask = (newTask) => { .....`